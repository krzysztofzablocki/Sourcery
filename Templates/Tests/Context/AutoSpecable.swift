//
//  File.swift
//  TemplatesTests
//
//  Created by Stijn on 22/03/2018.
//  Copyright Â© 2018 Pixle. All rights reserved.
//

import Foundation

protocol AutoSpecable { }

protocol ServiceProtocol: AutoMockable {
    func fetch()
}

struct BusService: AutoSpecable {
    // sourcery: testValue = ""Johny safely""
    // sourcery: customMock = ""Johny safely""
    let name: String

    private let service: ServiceProtocol

    init(name: String, service: ServiceProtocol) {
        self.name = name
        self.service = service
    }
}
