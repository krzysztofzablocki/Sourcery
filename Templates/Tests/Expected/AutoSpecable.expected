//: This code is not build or run. You can just copy paste it into your test file to have a start for the tests

/*
This code generates a spec template. Copy paste in to your spec file and rename class by removing the Auto.
Every test starts with sut (Subject Under Test) and its parameters that will be mocks.
For variables an methods you can use the following annotations:
* sourcery: typeNotMockable : will init the mock with the real value and init()
* sourcery: customMock = Mock() : will replace set mock to Mock() instead of TypeMock()
* sourcery: testValue : will add after initialization a test with == to this value like expect(sut.variable) == testValue (this also works for adding values to parameters)
* sourcery: skipSpec : will not add variable or function to spec
* sourcery: nestedVariableToTest : expect(sut.variable.nestedVariableToTest) == testValue
* sourcery: expectedReturnValue : expect(result) == expectedReturnValue
*/

import Quick
import Nimble
// @testable import TemplateTests


// MARK: AutoSpecs

// MARK: - BusServiceAutoSpec

final class BusServiceAutoSpec: QuickSpec {
  func testSpec() { spec() }

  override func spec() {
      describe("BusService") {
        var sut: BusService!

        var name:  String!

        beforeEach {
          name = "Johny safely"

           sut =  BusService(
  name: name
)
        }
        // MARK: - Test variables

        it("sets name") {
          expect(sut.name) == "Johny safely"
        }
      }
    }
}
